Source/SIMPLib/FilterParameters/DataArraySelectionFilterParameter.cpp

```diff
@@ -158,6 +158,39 @@ DataArraySelectionFilterParameter::RequirementType DataArraySelectionFilterParam
   return req;
 }
 
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+DataArraySelectionFilterParameter::RequirementType DataArraySelectionFilterParameter::CreateRequirement(const QVector<QString>& primitiveTypes, size_t allowedCompDim,
+                                                                                                        AttributeMatrix::Type attributeMatrixType, IGeometry::Type geometryType)
+{
+  using SizeTVectorType = std::vector<size_t>;
+  DataArraySelectionFilterParameter::RequirementType req;
+
+  for(const auto& type : primitiveTypes)
+  {
+    if(type.compare(SIMPL::Defaults::AnyPrimitive) != 0)
+    {
+      req.daTypes.push_back(type);
+    }
+  }
+
+  if(SIMPL::Defaults::AnyComponentSize != allowedCompDim)
+  {
+    req.componentDimensions = std::vector<SizeTVectorType>(1, SizeTVectorType(1, allowedCompDim));
+  }
+  if(AttributeMatrix::Type::Any != attributeMatrixType)
+  {
+    QVector<AttributeMatrix::Type> amTypes(1, attributeMatrixType);
```

Comment:
use `AttributeMatrix::Types` instead of `QVector<AttributeMatrix::Type>`.  That way, if we ever change the alias to another collection type, this will get updated as well.